stream_alert_cli/test.py

```diff
@@ -48,338 +56,493 @@ class TestingSuppressFilter(logging.Filter):
+        source = test_record['source']
+        compress = test_record.get('compress')
+
+        data_type = type(test_record['data'])
+        if data_type == dict:
+            data = json.dumps(test_record['data'])
+        elif data_type in (unicode, str):
+            data = test_record['data']
+        else:
+            LOGGER_CLI.info('Invalid data type: %s', type(test_record['data']))
             return
 
-    if service == 's3':
-        # Set the S3 object key to a random value for testing
-        test_record['key'] = ('{:032X}'.format(random.randrange(16**32)))
-        template['s3']['object']['key'] = test_record['key']
-        template['s3']['object']['size'] = len(data)
-        template['s3']['bucket']['arn'] = 'arn:aws:s3:::{}'.format(source)
-        template['s3']['bucket']['name'] = source
-
-        # Create the mocked s3 object in the designated bucket with the random key
-        put_mocked_s3_object(source, test_record['key'], data)
-
-    elif service == 'kinesis':
-        if compress:
-            kinesis_data = base64.b64encode(zlib.compress(data))
+        # Get the template file for this particular service
+        template_path = os.path.join(DIR_TEMPLATES, '{}.json'.format(service))
+        with open(template_path, 'r') as service_template:
+            try:
+                template = json.load(service_template)
+            except ValueError as err:
+                LOGGER_CLI.error('Error loading %s.json: %s', service, err)
+                return
+
+        if service == 's3':
+            # Set the S3 object key to a random value for testing
+            test_record['key'] = ('{:032X}'.format(random.randrange(16**32)))
+            template['s3']['object']['key'] = test_record['key']
+            template['s3']['object']['size'] = len(data)
+            template['s3']['bucket']['arn'] = 'arn:aws:s3:::{}'.format(source)
+            template['s3']['bucket']['name'] = source
+
+            # Create the mocked s3 object in the designated bucket with the random key
+            _put_mock_s3_object(source, test_record['key'], data, 'us-east-1')
+
+        elif service == 'kinesis':
+            if compress:
+                kinesis_data = base64.b64encode(zlib.compress(data))
+            else:
+                kinesis_data = base64.b64encode(data)
+
+            template['kinesis']['data'] = kinesis_data
+            template['eventSourceARN'] = 'arn:aws:kinesis:us-east-1:111222333:stream/{}'.format(
+                source)
+
+        elif service == 'sns':
+            template['Sns']['Message'] = data
+            template['EventSubscriptionArn'] = 'arn:aws:sns:us-east-1:111222333:{}'.format(
+                source)
         else:
-            kinesis_data = base64.b64encode(data)
+            LOGGER_CLI.info('Invalid service %s', service)
+
+        return template
+
+
+class AlertProcessorTester(object):
+    """Class to encapsulate testing the alert processor"""
```

Comment:
So I'm understanding this right, the alerts used as input come from the rule processor, right? can we make that clear in this classes docstring?